lib/vmpooler/api/v1.rb

```diff
@@ -121,90 +88,75 @@ def need_token!
+      end
 
-        if result[task][:count][:total] > 0
-          result[task][:duration][:average] = result[task][:duration][:total] / result[task][:count][:total] ##??
-        end
+      daily.each_key do |day|
+        result[:daily].push({
+          date: day,
+          boot: daily[day][:boot],
+          clone: daily[day][:clone],
+          tag: daily[day][:tag]
+        })
+      end
 
-        result[task][:count][:min], result[task][:count][:max] = daily_counts.minmax
-        result[task][:count][:average] = mean(daily_counts)
+      JSON.pretty_generate(result)
+    end
 
-        if daily_durations.length > 0
-          result[task][:duration][:min], result[task][:duration][:max] = daily_durations.minmax
-        end
+    get "#{api_prefix}/summary/:route/?:key?/?" do
+      content_type :json
 
-        daily_counts_bypool.each_key do |pool|
-          result[task][:count][:pool][pool][:min], result[task][:count][:pool][pool][:max] = daily_counts_bypool[pool].minmax
-          result[task][:count][:pool][pool][:average] = mean(daily_counts_bypool[pool])
+      result = {}
 
-          if daily_durations_bypool[pool].length > 0
-            result[task][:duration][:pool][pool] ||= {}
-            result[task][:duration][:pool][pool][:min], result[task][:duration][:pool][pool][:max] = daily_durations_bypool[pool].minmax
-          end
+      from_param = params[:from] || Date.today.to_s
+      to_param   = params[:to]   || Date.today.to_s
 
-          if result[task][:count][:pool][pool][:total] > 0
-           result[task][:duration][:pool][pool][:average] = result[task][:duration][:pool][pool][:total] / result[task][:count][:pool][pool][:total]
-          end
+      # Validate date formats
+      [from_param, to_param].each do |param|
+        if !validate_date_str(param.to_s)
+          halt 400, "Invalid date format '#{param}', must match YYYY-MM-DD."
         end
       end
 
-      result[:daily].each do |daily|
-        daily[:tag].each_key do |tag|
-          result[:tag][tag] ||= {}
+      from_date, to_date = Date.parse(from_param), Date.parse(to_param)
+
+      if to_date < from_date
+        halt 400, 'Date range is invalid, \'to\' cannot come before \'from\'.'
+      elsif from_date > Date.today
+        halt 400, 'Date range is invalid, \'from\' must be in the past.'
+      end
 
-          daily[:tag][tag].each do |key, value|
-            result[:tag][tag][key] ||= 0
-            result[:tag][tag][key] += value
-          end
-        end
+      case params[:route]
+        when 'boot'
+          result = get_task_summary(backend, 'boot', from_date, to_date, :bypool => true, :only => params[:key])
+        when 'clone'
+          result = get_task_summary(backend, 'clone', from_date, to_date, :bypool => true, :only => params[:key])
+        when 'tag'
+          result = get_tag_summary(backend, from_date, to_date, :only => params[:key])
+        else
+          halt 404, JSON.pretty_generate({ 'ok' => false })
```

Comment:
Makes sense. I thought this was a query param. `/summary?route=something` vs `/summary/something`.